examples/sampling/pyg/node_classification.py

```diff
@@ -182,52 +182,96 @@ def evaluate(model, dataloader, device, num_classes):
     )
 
 
+@torch.no_grad()
+def layerwise_infer(
+    model, args, infer_dataloader, test_set, feature, num_classes, device
+):
+    model.eval()
+    features = feature.read("node", None, "feat")
+    pred = model.inference(args, infer_dataloader, features, device)
+    pred = pred[test_set._items[0]]
+    label = test_set._items[1].to(pred.device)
+
+    return MF.accuracy(
+        pred,
+        label,
+        task="multiclass",
+        num_classes=num_classes,
+    )
+
+
 def main():
     parser = argparse.ArgumentParser(
         description="Which dataset are you going to use?"
     )
     parser.add_argument(
         "--dataset",
         type=str,
-        default="ogbn-arxiv",
+        default="ogbn-products",
         help='Name of the dataset to use (e.g., "ogbn-products", "ogbn-arxiv")',
     )
+    parser.add_argument(
+        "--epochs", type=int, default=10, help="Number of training epochs."
+    )
+    parser.add_argument(
+        "--batch-size", type=int, default=1024, help="Batch size for training."
+    )
     args = parser.parse_args()
+
+    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
     dataset_name = args.dataset
     dataset = gb.BuiltinDataset(dataset_name).load()
     graph = dataset.graph
-    feature = dataset.feature
+    feature = dataset.feature.to(device)
     train_set = dataset.tasks[0].train_set
     valid_set = dataset.tasks[0].validation_set
     test_set = dataset.tasks[0].test_set
+    all_nodes_set = dataset.all_nodes_set
     num_classes = dataset.tasks[0].metadata["num_classes"]
-    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
 
     train_dataloader = create_dataloader(
-        train_set, graph, feature, device, is_train=True
+        train_set,
+        graph,
+        feature,
+        args.batch_size,
+        [10, 10, 10],
```

Comment:
```suggestion
        [15, 10, 5],
```
Should we use same fanout values as the PyG original example?
https://github.com/pyg-team/pytorch_geometric/blob/25b2f208e671eeec285bfafa2e246ea0a234b312/examples/ogbn_products_sage.py#L20-L28

Reply:
I am not sure if the ordering of fanouts is the same as PyG's though.